### ISSUE TITLE
Blade of Agony videos

### ISSUE THREAD

**Issue by Username-N00b-is-not-available** [Realm667/WolfenDoom#281]
↑↑↑ Resurrecting the issue ![resurrect](https://user-images.githubusercontent.com/59409713/102812424-09b84400-43d8-11eb-89c3-1687e7ead574.png) ↑↑↑
---
@Tormentor667 @Ozymandias81 @Talon1024

I have recorded another full playthrough. This time I played on the lowest skill and saved extensively (since the crashes have been apparently dealt with), so it is shorter than the previous edition (11 parts vs 14 on C1, 26 vs 40 on C2, and also 26 vs 37 on C3). I have now included the Commander Keen levels too. I used GZDoom 4.6pre20 and this BoA commit: https://github.com/Realm667/WolfenDoom/commit/518865ee16c6d952edc283cc9723088c018c635e

Hopefully I will have it fully uploaded in a few days. For now, there is only the first episode. If you do decide to watch some of these, please use the speed multipliers, because I'm not a speedrunner, and the footage could become boring to watch.

Overall I think the game is also fine on this skill, there are still many nitpicky problems on maps here and there (including some very old ones), but everything is playable, finishable and fun to blast through as a power fantasy. What stood out from this list of problems in my mind:

- [x] The soul barrier before the final room in C3M6_B has a blocking linedef which hinders the player's ability to move through it, even if there are enough souls (but after a few tries you can pass it anyway). _[assigned to @AFADoomer]_

> I cannot reproduce this. - AFA (9 Jan 21)

> Here is a video of this, commit https://github.com/Realm667/WolfenDoom/commit/212a3d9922cbce6183450bcdb10ee584035f8c8f: [barrier.zip](https://github.com/Realm667/WolfenDoom/files/5795560/barrier.zip) --N00b, 11 Jan 2021

> OK, should be good to go as of f9d717a55c7f0b3d561137c3ab5bc45a3ff243a5. - AFA (12 Jan 21)

- [x] In the final battle Hitler is invisible again. I seriously do not know what could have been the cause of this, played a bit with the settings after encountering this, but could not make Hitler show himself. He should also be instantly awakened after spawning and pointed at the player (as for now, he is not). _[assigned to @Talon1024 @AFADoomer @Ozymandias81]_

> @Ozymandias81 You have tried to fix these problems some days ago https://github.com/Realm667/WolfenDoom/issues/303#issuecomment-751498040, but they are still there (except the spear spawnheight), I can provide a video if needed. --N00b, 1 Jan 2021

> It seems that these problems have been fixed. --N00b, 11 Jan 2021

Next portion of issues which need help (resulting from @Tormentor667 watching my videos):

- [x] @Talon1024 There are some missing characters (Cyrillic ones and probably dashes) to the font used by the Commander Keen dialogues, could you add them please?
- [x] @AFADoomer Peeing enemies (and probably not only them) do not react to gunshots of their comrades: https://www.youtube.com/watch?v=UbaGwv7CWmc&t=0m1s
> I thought that this was intentional?  They are peeing too loud to hear 😁 .  Can be fixed by doubling/quadrupling frames and adjusting the random tic amounts downward accordingly. - AFA (30 Dec 20)
- [x] @Talon1024 Could you please reorganize the C3M0_A Nemesis behaviour, so that the final fight starts exactly upon the player exiting the SS HQ? It would also be more consistent if the Nemesis does not enter the HQ ('hiding' somewhere in the streets). What happens now: https://www.youtube.com/watch?v=PrlVqdQ_HoM&t=9m10s (you just have to wait until he respawns again).
- [x] @Talon1024 @AFADoomer Final Ryan battle is useless: https://www.youtube.com/watch?v=m8vmFpKu5Vk&t=3m10s

Added 20 Jan 2021 (no video material)
---
New C1 playthrough on standalone boa.ipk3 version:
- [x] Can't skip Douglas typewriting sequence in its middle.

> Works for me.™ - AFA (21 Jan)

> LOL, whatever. Have you tried skipping it closer to the end of the sequence? @Talon1024, @Tormentor667, maybe you have this problem? --N00b, 22 Jan 2021

- [x] Devcomments should be on by default, they aren't.

> Why would they be on by default?  They would distract from gameplay for first-time players - AFA (21 Jan)

> I agree with you, but the first devcomment, DEVINTER_10 by @Talon1024, says '...If you leave these developer comments turned on...' --N00b, 22 Jan 2021

- [ ] In some Strife dialogues, especially in the HQ, the mugshot animations are too fast, the frames probably change every tic or so.

> I've already slowed these down once.  Maybe this is machine-dependent...  Will check. - AFA (21 Jan)

> This still happens with NPCs including the surrendering soldier on C3M5_A. --N00b, 07 Feb 2021

- [x] On the INTERMAP shooting range, it is possible to hit one of the rifle targets (the higher one) from the pistol shooting position (the left one is blocked by a tree). The barbed wire linedefs should block projectiles.
- [x] INTERMAP: the slide projector has its spherical light lower than it should be:
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_123028](https://user-images.githubusercontent.com/59409713/105149832-0a4a1480-5b15-11eb-8875-2507f6b33c23.png)

</details>

- [x] Add a larger (random?) delay before wounded soldiers raise to shoot at the player (they now do it almost instantly).
- [x] C1M1: this dog often gets stuck:
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210118_195319](https://user-images.githubusercontent.com/59409713/105149214-5a74a700-5b14-11eb-87ba-8eeb5f349731.png)

</details>

- [x] Cooks from the Chef Team use new active sounds for Dr. Schabbs.
:x:  This corpse glows:
> Nothing strange, bloom effect augment lightpoints in place , and if you don't have bloom on then it is just how lights works on GZDoom. No brightmapre are present for the actor too. --Ozy81
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_111519](https://user-images.githubusercontent.com/59409713/105149717-e8509200-5b14-11eb-8baf-83b920941627.png)

</details>

- [x] C1M2: we should lower the technical sector in the Opel Blitz car with the secret hint sheet, the paper floats on air (you can search for doomednum 32122 in UDB):
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_103318](https://user-images.githubusercontent.com/59409713/105149438-97d93480-5b14-11eb-8777-279f122cfd1b.png)

</details>

- [x] C1M2: Ryan's message about triggering the alarm is played too late.

> What do you mean?  The ACS script that sets off the yellow strobes and alarms (normally when you pick up the MP40 in the red key area) is literally the same one that makes him print his message. - AFA (21 Jan)

> Maybe, but in my playthrough the warning bell in the room just opposite to the mess hall (wrt the main long passage) was for some reason activated just after I had rescued Ryan (and this also spawned a squad of 3 SS MP40 guards in the open cell near the bell). The bell can not be deactivated, so I must have mistaken it as a general alarm. --N00b, 22 Jan 2021

- [x] C1M2: got 236 kills of 235 (bats?), and the ending APC does not count towards kills for some reason.
> Bats don't count toward kills at all anymore...  Also, just did a playthrough and got 100% kills.  Fixed the Tank kill count issue in b4af7ed75b9bbf7023185f479601dc1a39e1d99c. - AFA (21 Jan)

> Well, the first issue is likely irreproducible, so thank you. --N00b, 22 Jan 2021

- [x] C1M3: looks like the invisible turrets in the main trench area have disappeared somewhere? Not that I miss them. Was it a design choice?
- [x] On skill 2 (default) enemy grenades damage other enemies, is this intended?

> They have never *not* hurt other enemies.  - AFA (21 Jan)

- [x] C1M3: mirror in room (9106, 1120, -16) plays switch sound on being broken.
> Can't reproduce this, everything seems correct... Just in case I have included a SOUND def under animdefs. --Ozy81
- [x] C1M3: these two soldiers are easily missed during a walkthrough, they don't get alerted if the player does not visit this trench. I'd place two more soldiers around that trench, but on the high ground, so the player goes to loot their bodies and encounters those two.
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_122457](https://user-images.githubusercontent.com/59409713/105149760-f4d4ea80-5b14-11eb-9960-d8e66037670e.png)

</details>

- [x] C1M4: maybe we shall move the secret hint sheet to some other location? I've placed it on a crate, but maybe it is better to move it where it can be found more easily (again, search for 32122)
- [x] C1M4: the question mark sign clips through the Commander Keen safe. Also seen with two other safes, on C2M4 and C3M3_A.
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_125922](https://user-images.githubusercontent.com/59409713/105149886-1930c700-5b15-11eb-962b-0e3b92b50600.png)

</details>

- [x] You can activate another bandage before the first runs out, and their effect does not stack. Is this deliberate?

> No.  Fixed in d0eede109fed4e1604d160ac7c7828df68adae43. - AFA (21 Jan)

- [ ] C1M5: there is absolutely no punishment for the Nebelwerfer being destroyed, I can Panzerschreck it during wave 8 and still get all credit for the mission. Maybe a $250 fine would do?
- [x] Improper Totale Gier rewards from some bosses (noticed it a bit earlier, when I was recording Graf Holzbein's sounds in C2M3; on death he dropped three or four coins).
- [ ] Commander Keen messages can be skipped by any key including the movement ones, possibly should only leave 'use' for this.
> This matches every other fullscreen message/menu interface in the game right now.  Is the real problem that message pages are accidentally being skipped? - AFA (23 Jan)

> Yes, you are right, that's what I meant. --N00b, 27 Jan

- [x] No indication on Commander Keen doors, <strike>I'll think of a crutch to deal with it myself.</strike>

> Looks like @Tormentor667 had left the 'Not shown on map' flag on the doors' linedefs :-D --N00b

- [x] This Mushroom is dormant and floats in mid-air:
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_190623](https://user-images.githubusercontent.com/59409713/105149926-264db600-5b15-11eb-8018-0dc754cf9966.png)

</details>

- [ ] On normal skill, please remove target icons which are shown on snipers. They are not hitscan, and the player can already see the bullet trails.
- :x: C1M0: main lift has 3dfloor railings which cannot be shot through, later I'll try replacing them with walkable midtextures like in other parts of the level.

> Forgot that we can't easily tie a walkable linedef to a 3dfloor vertically. --N00b

- [x] C1M0: Z-fighting over and under newly-added main lift buttons. I forgot this problem long ago...
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_193653](https://user-images.githubusercontent.com/59409713/105149955-2f3e8780-5b15-11eb-8072-c0f157a70880.png)

</details>

- [x] C1M0: the gas effect after 25min have passed instantly disappears after bombing the crates, if there is a way to have it stay applied, it would be nice.
- [ ] Suggestion: show the mission summary screen (the one which is in the briefing room in the HQ) after a chapter playthrough?
- [x] C1M6: you are not given the ArtifactEgyptian in the beginning of the mission + it should be taken from your inventory on using it to open the gates.
- [x] C1M6: in its first encounter (in the main hub of the pyramid, long before the final battle) the smoke monster can not be seen.
- [x] C1M6: please remove the Holy Grail from item count, it is the only item which prevents 100% items.
- [x] C1M6: these walls near the Grail room need some blocking lines, you can walk inside them:
<details><summary>Screenshot</summary>

![Screenshot_BoA_20210119_210828](https://user-images.githubusercontent.com/59409713/105149988-38c7ef80-5b15-11eb-8385-583e60ccd9a7.png)

</details>

- [x] C1M6: the last 12 soldiers who spawn after the boss fight should be removed from countkill. Before the final fight I had 121/121 kills, after it 122/134.

Added 22 Jan 2021 (no video material)
---

- [x] C2M1: you can safely shoot from this vent and break the alarm without being noticed by guards (or shoot one of the guards: if he does not manage to shoot before his death, the other one will not be alerted).

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210120_131151](https://user-images.githubusercontent.com/59409713/105542157-f74c6580-5d09-11eb-8b31-f2e3cc826d85.png)

</details>

- [x] C2M1: execution schedule has inconsistent line spacing (this is a reminder for myself to improve it in all languages a bit later).
- [x] C2M1: barely enough ammo to kill guards on all watchtowers, a few 9mm and 7.92mm clips dropped along the path would be appreciated.

> Added two 7.92mm clips and several 9mm ones in 'risky' places --N00b

- [ ] C2M1: The alarm activates upon entering the light even if all the guards on the tower are killed.
- [x] C2M1: no invisible turrets in flamer trench area, were these also intentionally removed? Same question for the direct path to the port in C2M6_B.
- [x] C2M1: 212/221 kills. I checked everything, every area, including the watchtowers. In that state, when I ```kill monsters``` the game kills exactly one actor (Douglas). This is probably because of guards who despawn around the time you get the blue key.
- [x] C2 intermap chest again clips slightly through the briefing room wall.
- [x] C2M2: departing train sound effects are repeated each time you cross the activating linedef near the red door.
- [x] C2M2: sniper at (5344, 4528, -704) just sits in the dark and does not even shoot at the player. This is a really old issue, I remember having reported it multiple times.
- [x] C2M2: the secret hint sheet floats in midair (around 32 or 40 units from floor).
- [x] C2M2: mecha battle can be cheesed: you have enough time to get on the upper ground and alert them one by one, and because of their low pain chance they could be plinked at from corners without even being alerted.

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210120_155139](https://user-images.githubusercontent.com/59409713/105542205-03d0be00-5d0a-11eb-83b1-b0d6516c2115.png)

</details>

- [x] C2M2: Ryan in ending sequence is still thrust by grenade blasts.
- [ ] Lt. Heit shows a message about the player completing Schwarzwald before C2M3.

> From what I can see, Randi is supposed to use lines BARRANDICONV01 through 06 before C2M2, then nothing new before C2M3, then 07 before C2M4, 09 before M5, 10 before M0, and finally 11 before M6 if M0 was completed. BARRANDICONV08 about "the Thor's Hammer and the long d*ng" does not have a place to fit in. --N00b

- [x] Jukebox music resets to default orchestra cover of a Wolf3D midi after returning to intermap, is it intended?
- [x] C2M3: there are gaps between the upper parts of walls and the roof in most houses. Probably a quirk of the 3dfloor plugin.

> This was not a big deal for the most of houses, except the purple key mansion. I improved it a bit. --N00b

- [x] C2M3: cyan door at (7412, -344, -368) can be opened from below by pressing the wall.
- [x] C2M3: door at (4248, -684, -182) has improper texture offsets.
- [ ] C2M3: ladder (9159, 6842, -91) has same problem as the Loper lab one in C3M6_A (it throws the player upwards and out of the portal if they touch one of its sides).
- [x] C2M3: misalignment on crate (14137, 2796, -272).
- [ ] Under Totale Gier, enemies do not drop gold if they die near walls.
- [x] C2M3: the Hitler's Ghost in area (13805, -4139, -136) can be alerted prematurely, from the dog ambush area to the north.
- [x] C2M4: starting area soldiers are killable (including Ascher).
- [ ] C2M4 suggestion: add a reward for destroying all planes?
- [x] Lower Ubersoldat pain chance and/or lower the pain state time? They are too vulnerable against shotguns.
- [x] C2M4: you can get inside the car 3dmodel (its front side) at (15540, -4124, -88).
- [x] C2M4: please add -COUNTKILL on 26 soldiers who get spawned at the end, it is not feasible to kill them all in time with the Fliegerfaust's fire rate.
- [x] C2M4: why does BJ get off the truck in the ending sequence, is he going to travel the remaining part on his own?
- [x] WTF on CUTSCEN2, very likely something with Motion Blur or related shaders. I have never seen it before. As you can see, the strange effect does not wear off with time, and it occured for me also on C2M0_A and C2M6_C (but there was one normal remaining cutscene, I forgot which). The only renderer parameter I have changed for this playthrough from a clean installation was cl_capfps, but I doubt this is the source of the problem, since I have always played with it, and everything was OK.

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210120_191340](https://user-images.githubusercontent.com/59409713/105542280-219e2300-5d0a-11eb-91ec-849441bb646b.png)

</details>

- [x] Russian text in CUTSCEN2 has an empty page at the end, why? Also true for all other 'blue screen tally ending' texts which end on the last line of the page. I guess that the code which counts the number of pages required to show the lines has an off-by-one error.
> Fixed in 410c670bac0c784bd07f1d5a7e58e9c56abdca9e. - AFA (23 Jan)
<details><summary>More evidence (from C3M0_A): the encircled line appears on two pages but should only be on one</summary>

![linenum](https://user-images.githubusercontent.com/59409713/105576921-51d7d700-5d87-11eb-8113-9ac107a619e3.png)

</details>

> Should be fixed in 5db2651b9b571f87a56b8b74df560f2cf2bcc6c0. - AFA (23 Jan)

- [ ] Should not allow to buy Kar98k from the shop before C2M5 (you are given one).
- [ ] A problem with devcomment texts: text colouring sometimes starts at the wrong point. The screenshot has the opening parentheses coloured orange, while the colouring in the text starts after them (an excerpt from the CSV for the first one: ```...используя ZFontConverter (\cIhttps://github.com/Talon1024/ZFontConverter\cC), между...```) Same problem can be seen with the Feld Hell devcomment and some others.

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210120_191521](https://user-images.githubusercontent.com/59409713/105542337-3d092e00-5d0a-11eb-9977-a52facc12bb5.png)

</details>

- [x] C2M5_B: I regret my decision to tie the door to area (3984, 12895, 548) with the local entrance door, it does not seem obvious to me at all now. Please make them open separately...
- :x: C2M5_B: I'd place walkable linedefs instead of 3dfloors on moving lift (just a nitpick).
- [ ] C2M5_B: the zaps do not work properly, the player can pass through most of them without receiving any damage.
- [x] C2M5_B: 279/281 kills (because of MGTurret near (2957, 4935, 532) and Douglas, who has +COUNTKILL for some reason). MGTurret flags after deactivation:
<details><summary>Screenshot</summary>

![mgturret](https://user-images.githubusercontent.com/59409713/105542138-ec91d080-5d09-11eb-8a26-5cfa0e849c35.png)

</details>

- [x] It would be very nice if we can raise the default autosave amount to 20 and mark technical maps such as C3INTRO and CUTSCENx with this in the mapinfo definition: https://zdoom.org/wiki/MAPINFO/Map_definition#NoAutosaveHint
- [ ] C2M0_A: full path to image in archive is shown in the upper part of the screen when seeing Kerner's document.
- [x] Totale Macht does not protect from toothpicks and boulders (it only manages to activate after the player has already died), though not sure if we should consider this as a problem. Also occurs with Astro Cyborg projectiles and other (but not reliably all) high-damage attacks.
> Changed the Totale Macht in 268a345af428624aafda6e0b10c0830be60ea12b to protect against even telefrag damage.  Still no real change with floor spikes, because they continue to damage at a rate higher than the regen can restore, but should make the response to other damage more reliable. - AFA (24 Jan)

- [x] C2M0_A: boulders near (16819, 7869, -254) stack over the instant death floor.
- [ ] C2M0_B: you can abuse vertical airlock doors this way, not sure if it is important, but as a nitpick I would like to have some Sector_SetLink on all of them.

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210121_181536](https://user-images.githubusercontent.com/59409713/105542388-514d2b00-5d0a-11eb-9852-d09636086312.png)

</details>

> Reduced the problem to this screenshot, no idea why the sloped floors behave this way. @Talon1024 @Tormentor667 Could you please help me? --N00b, 12 Feb 2021

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210212_194140](https://user-images.githubusercontent.com/59409713/107796228-abba3400-6d6a-11eb-8e8f-dd393cdb2d1b.png)

</details>

- [x] C2M0_B: I'd deduct armor shards from item amount. A Doom tradition is nice, but from small items we only have gold that counts towards items.
- [x] C2M0_B: these scientists shoot their pistols through windows, and bullets fly through them (my reflexes did not allow me to catch the tracer on a screenshot). Also I've spotted one or two times that tracers continue living (and flying) after the bullet hits something; I remember this happen on a one-sided linedef in C1M1's Rick's Hotel Americain (the SS officer shot at me, but hit the wall with a very small angle from its surface to the bullet's trajectory, and the tracer continued travelling and 'hit' me, but of course with no damage).

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210121_183352](https://user-images.githubusercontent.com/59409713/105542411-5a3dfc80-5d0a-11eb-934a-e3244d49255b.png)

</details>

- [x] C2M0_B: It was recently added that a soldier who looks in the mirror can see the player going from behind and gets alerted... but this does not work in Astrostein, example: (-11872, 8096, 346).

> Single occasion, not worth it. --N00b, 12 Feb 2021

- [x] C2M0_B: Quad-barrel mecha guard at (-13438, 4602, 282) is not alerted by gunshot sounds and cannot see you if you are around (-12864, 4736, 282).
- [x] Astrostein barrels do not play any death sound.
- [x] C2M0_B: 275/276 kills because of unused thing 484 @ (2912, 4224, 48) (the cyborg is superseded by one that gets spawned by script 7, we should remove him).
- [ ] BARRANDICONV11 is never spoken by him (after completing Astrostein he still has his 10th message).
- [x] C2M6_A: in the complex enemies can dodge into a 32-unit niche and get stuck (I also suspect this about C2M1 and C2M5_B).

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210122_120742](https://user-images.githubusercontent.com/59409713/105542454-6e81f980-5d0a-11eb-90c8-93523baac699.png)

</details>

- [x] C2M6_B: unbreakable mirror near (7152, 11260, -32).
- [x] C2M6_B: Soldiers at (6500, 2700, 144) keep speaking after death, same for (4836, 3030, 16), (1176, 136, 32), ...
- [x] C2M6_B: door near (2040, -560, 192) has improper texture offsets.
- [x] C2M6_B: impassable linedefs in the submarine which represent closed doors do not block projectiles, see (4232, 9357, -128).
- [x] C2M6_B: got 177/184 kills. Unreachable enemy troops in purely decorative areas, such as (5130, 2017, 144), (5130, 2933, 144), and (6923, 3005, 400) should not count towards kills, and I think it would be better to remove Seeteufel from this map altogether, so that Seeteufel and Giftmacher come aboard together, outside of the player's view. This way you can reach 100% kills (yes I know this is not intended, but there is so little to be done to not alienate the completionists or max-speedrunners).
- :x: Add ability to patch the heavy armor with Stahlhelms over 100%, up to 200%?

> [No], I'll leave it to mods. Should not be too difficult with ZScript. --N00b, 12 Feb 2021

- [ ] One of the textures used in C2M6_C has 'Projekt Seeteufel' on it, maybe it is better to either rename the enemy or change the project name to https://de.wikipedia.org/wiki/Schwertwal_(U-Boot) so as not to confuse the players?
- [x] C2M6_C: adrenaline objective should be marked as accomplished when you enter the storage room, not when you pick it up, same as for Totale Macht in C2M6_B.
- [x] C2M6_C: Douglas near (9392, 11284, -72) can be thrust by grenade/flamer explosions.
- [x] C2M6_C: Sd. Kfz. 223 should be spawned as active, it just stands in the player's plain sight when they have not yet left the courtyard of the building with two Nebelwerfers on top.
- [ ] C2M6_C: same message for Douglas as transmission and in dialogue in the excavation.
- [ ] C2M6_C: 249/248 kills, no idea why.

Added 25 Jan 2021 (no video material)
---

- [ ] STATS_NONE is not shown on the stat text screen in C3 (not sure about two other chapters), there is just empty space.
- [x] C3M1_A: please remove +countkill on the firing squad or teleport them somewhere instead of despawning.
- [x] C3M1_A: do not despawn soldiers on bombing the storage, either kill them, teleport somewhere, or launch Thing_SetGoal on those who are inside the camp area.
- [x] C3M1_A: please raise brightness in Mengele basement a bit, it is very dark there.
- [ ] C3M1_A: 132/135 kills (I did not blow up the storage, killed all the guards instead).
- [ ] We should check that a knife is given after returning to intermap from C3M1_A.
- [ ] C3M1_B: SS MP40 guard near a mirror at (1715, -1438, 433) does not see BJ sneaking behind him.
- [ ] C3M1_B: 92/93 kills, probably something with Ascher (who can be thrust by explosions), same applies to C3M1_A.
- [ ] C3M0_A: killcount is totally off, it may be better to disable showing it at all. Pre-placed zombies count towards kills, spawned ones not, and it is impossible for a player to distinguish them.
- [ ] C3M0_A: we probably should use SpyGive-like behaviour for bonus items (drop the surplus items near the player). That will likely mean migration from ACS.
- [x] C3M0_A: house at (-4753, 5531, -184) has breakable windows which still block players after being broken, and there is a gap between the roof and walls.
- [ ] C3M0_A: monk near (12154, 3850, -224) does not see the player. Same with Gen. Totengraeber.
- [ ] C3M0_A: can't reach loot from barrel near (5442, 20064, 768).
- [x] C3M0_A: improper scaling on wooden textures behind the breakable mirror in Arlow's bathroom.
- [x] C3M0_A: you can take the secret UMG43 without triggering the secret.
- [ ] C3M0_A: absolutely nothing indicates that to start assaulting the SS HQ you have to gather all three weapons (Browning, G43, Nebelwerfer) from their specific spots. Else the player can only wonder what to do after grabbing two keys and, for example, missing the butcher's with the rifle. I think the corresponding primary objective should be re-added.
- [x] C3M0_A: when Arlow contacts the player via radio he has a spy mugshot.
- [ ] C3M0_A: got 4 of 5 treasure items (all of them in the cinema), can't find the last one.
- [ ] C3M0_A: Is it really necessary for the player to wait several minutes until the Nemesis wakes up again after destroying the SS HQ? Could you please force Nemesis to revive himself when the player leaves the area, or at least give another message the player to wait a bit 'until the reinforcements arrive'?
- [ ] C3M0_A: I have severe lag in the city part of the level, I have to pause the game every few seconds to avoid stuttering. Could anyone please think of some measures to reduce the workload on the CPU, maybe adding more one-sided linedefs which block the view, or replacing lots of small decorations with bigger ones?
- [ ] C3M2: the Allied paratroopers can be killed by player's weapons.
- [ ] C3M2: the map is lagging as hell, including during the machinegun hill defending scene and the tank ambush (where it is especially frustrating), both on default and low settings. On default my FPS does not raise over 15 and frequently drops to 5. Any measures here as well?
- [ ] C3M2: crates at (3870, -308, 224), (2688, -1255, 312) have flat misalignment.
- [ ] Medics are useless, they get activated only when the player comes close (and shoots them, of course), they need to roam more freely to be a threat (this is related both to map placement and actor behavior). This is not only true of C3M2, but all maps which use medics.
- [ ] C3M2: On default settings, because of the post-processing effects the tank ambush plays awful. You can't see the Pz. IV's properly because of that shaders and BS fog, not to mention that they cause additional workload on the GPU when it's already lagging excessively. Also the Pz. IV's have more health than Tigers (but still go down in two main cannon shots).
- [ ] C3M2: the two Allied soldiers near (8736, -6890, 0) are unable to move after being told to stay at that location and later told to go with Blazkowicz.
- [ ] C3M2: most of the Allied soldiers in the ruined mansion do not react to German troops who are trying to attack it, and their bullets damage the player more than do any harm to Germans. There is also huge lag in the area.
- [x] C3M2: sniper above (13284, -479, 56) cannot shoot at the player and is very unlikely to be spotted, making 100% kills a problem; he should be removed or moved to another location.
- (not a bug) Got exactly 100% kills on C3M2, very nice work! Thank you for accomplishing this!
- [ ] Arlow's dialogue in the HQ should be removed after C3M2.
- [ ] Would it be appropriate to remove the 'Operation' word from C1M1 through C3M0_B entries of the CSV (they hold the map names)? Some of them are too long and overlap with the playing time in the INTERMAP stats screen.
- [ ] C3M3_A: a dog can get stuck near (4005, 6123, -103) and not get out to chase the player.
- [ ] Is it really necessary to have a damage penalty for unscoped Kar98K fire?
- [x] C3M3_B: what does the switch near (16964, 5632, 8) do? Does it reopen the main door 'just in case it closes'? Why is it set to one-time activation (or is it?)
- [ ] C3M3_B: area (16791, 6228, 136) with three SS MP40 guards is supposed to open when the player crosses the bridge, but the activating linedef also triggers on being crossed under the bridge.
- [ ] C3M3_B: 80/81 items, can not find the missing one. I searched literally every corner with double iddt + editor, no idea where (and what) it could be.
- [ ] C3M4: please remove countkill on the pre-placed zombies. There are no problems and you can get 100%, but to a first-time player it would be very non-obvious that the zombies on the ramp across the river count towards kills.
- [x] C3M4: coin at (-3684, -6584, -56) is spawned in the railing, you cannot see it. It should be moved to the ground.
- [ ] C3M4: 145/146 items, can't find the last one (I've successfully taken the coin from the previous checkbox).
- [ ] C3M0_B: I reached a state of the map when no enemy soldier in the complex could see me, even in bright light and not crouching. I could also kill everyone with projectile weapons, and nobody would get alerted. This started from the two dogs outside the complex (near the flares) who should chase me, they did not.
- [ ] The previous point allowed me to see that sneakable actors (which use +FRIENDLY) can be damaged and killed by alerted ones, even on higher skills.
- [x] C3M0_B: MGTurrets near the entrance have +COUNTKILL, but you're supposed to deactivate them, not destroy.
- [x] C3M0_B: polyobject vault door should crush the player, or you can end up as such:

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210124_164743](https://user-images.githubusercontent.com/59409713/105759826-f9a8fc80-5f61-11eb-8c03-b8bfbf1f3774.png)

</details>

- [ ] C3M0_B: final 17 enemies who spawn should not have +COUNTKILL. No other problems with kill count.
- [ ] C3M0_B: VM abort after the final sequence fades out:

<details><summary>Screenshot</summary>

![crash](https://user-images.githubusercontent.com/59409713/105759824-f9106600-5f61-11eb-8096-c880ffec1c80.png)

</details>

- [ ] C3M5_A: Allied soldier companions (except Ascher) are easily killed by player's tank machinegun fire.
- [ ] Need to justify offsets for machine gun effect spawners on various tank models.
- [ ] C3M5_A: Most of the time Ascher is closer to Mengele than the player, and so Mengele only attacks Ascher.
- [ ] C3M5_A: you can run past one of the four bosses (e. g. Goebbels) and start fighting the next one (in this case Morell), then upon killing the latter Ascher will say you have killed the former. You can even just run past each one of the four, not engaging in combat with them at all, and still proceed through the mission.
- [x] C3M5_A: officers can roll under the table at (-10099, -5880, -344) and stand up inside it.
- [ ] C3M5_A: toilet officer at (-15751, 14912, -2) disappeared on being shot with STEN altfire.
- [x] C3M5_A: Ascher can pass through grate near (-19060, 12691, -508) and similar ones.
- [ ] C3M5_A: Wehrmacht enemies in the underground are alerted of you when you pass through the exit door of the previous boxcar. You can open both doors without doing it, throw a grenade when they are still dormant, and wake them up when it is about to explode.
- [x] C3M5_B: the first secret now only gives three souls.
- [x] C3M5_B: doors in room (-8228, 1380, 128) can be opened from the 3dfloor above them.
- [ ] C3M5_B: some problem with overlapping portals (you can see a Luftwaffe officer from other area):

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210125_131150](https://user-images.githubusercontent.com/59409713/105759851-04639180-5f62-11eb-82f7-d08d6777d73e.png)

</details>

- [x] C3M5_B: please remove my cutscene area attempt at (-12000, -6000), it is not needed anymore.
- [ ] C3M5_B: 164/168 kills, I don't get at all where the remaining enemies could be. Triple-checked the automap and the editor, no clue. No moving enemies on map, no standing ones. If it helps, if I do 'kill Nazi' in the console after that I can hear Hans Grosse's death sound.

> And now the death sound is 'Scheisskopf!' --N00b, 07 Feb 2021

- [ ] C3M5_C: saluting SS guards do that even after being alerted of the player.
- [ ] C3M5_C: there is apparently a way I can make soldiers not react to gunshots and seeing the player with a weapon in the luxury interior area. After that I can just go around and kill everyone with the Blade.
- [ ] C3M5_C: Ascher's 'talking head' in SS uniform in the radio messages does not play its animation.
- [ ] Stealth system counts enemies who have already surrendered as ones who can still see you (not a serious problem, you can wait until they disappear).
- [ ] C3M5_C: 54/55 items.
- [x] C3M6_A: player can see these dogs in their dormant state:

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210125_163443](https://user-images.githubusercontent.com/59409713/105759892-0fb6bd00-5f62-11eb-8ac0-ad3dd35b8857.png)

</details>

- [x] C3M6_A: lamps and flags above (-6024, -4384, -3207) block almost all bullets, it makes fighting the Ubersoldat a problem.
- [ ] After seeing a spy shoot at real enemies Ascher opens fire upon him too:

<details><summary>Screenshot</summary>

![Screenshot_BoA_20210125_164826](https://user-images.githubusercontent.com/59409713/105759958-29f09b00-5f62-11eb-85dc-7f4516bd0661.png)

</details>

- [ ] Rapid-fire actors, such as controllable machinegun turrets, and even Ascher, continue shooting at their target even if a polyobject door between them has closed (but they won't start a new salvo in such circumstances).
- [ ] C3M6_A: can't buy the castle map from Hosenscheisser, it says "You don't need it". I don't even have any text paper in the inventory.
- [ ] C3M6_A: can buy the suspicious key multiple times (but only one of them will be in the inventory), wasting money.
- [ ] C3M6_B: the stand for the Spear of Destiny below (-5753, 4335, 168) has z-fighting on its upper parts (visible from top).
- [ ] C3M6_B: Hitler does not count towards kills. Or should he?
- [ ] ddb_crd2/ddb_crdt (graphics/intromap) are quite old (and still have some funny errors which have been fixed in texts long ago, such as Scuba and Steve in the playtesters section).
- [ ] If the player has no souls, he is able to power up the Firebrand with his own HP. That's fine, but this way they are still able to use the altfire attack, and it would cost them nothing until the effect runs off and the player has to use up their HP again. Is this tolerable or not?
- [ ] The first infopage has a screenshot of a map in some C3M6_B location which has changed (Shermans in the Soviet forces were changed to T-34s).
- [ ] We probably have to make a final <strike>solution</strike> decision for the sprinting and flinch amount in Cheats Options menu if we want to see BoA speedruns. These settings have a big impact on gameplay, and the player is free to change them as they are not considered cheating in-game.

Video links
---

<details><summary>Chapter I</summary>

https://www.youtube.com/playlist?list=PLPn07arnrFIKme92teVJRsJRwO48-oruc

</details>

<details><summary>Chapter II</summary>

https://www.youtube.com/playlist?list=PLPn07arnrFIIFu_YFG4CWvJ01MnNNTvIU

</details>

<details><summary>Chapter III</summary>

https://www.youtube.com/playlist?list=PLPn07arnrFIIsPcWrdTmb1fwbK5Z9B-ny

</details>

**FIRST PLAYTHROUGH FROM AUGUST**
<details>I'd like to finally record myself playing through the episodes and share the recordings, they will be hopefully more useful as videos instead of the demo format I used before.
My goal is to playtest the maps on Mein Leben difficulty, hopefully most of the three episodes, continuously with saves. I know that @Tormentor667 is currently playtesting the maps and checking them for bugs, and will therefore try to present the maps from the gameplay/resource management/story/etc. perspective (but of course still point at problems I find). I am not very good at Doom, but since I know the maps quite well, the show shouldn't be exceptionally painful to watch :-)
From a technical perspective, I am going to record 1920x1080 with a 640x360 canvas @ 30 fps using a bitrate of 600+96kbps.
If this effort seems useful to the team, tomorrow I will try recording the videos and then uploading them slowly, one by one (my internet connection is, again, bottlenecking the process). I will then also post the game settings I use for recording. So, does the team need this now?

<details><summary>Playthrough</summary>

[Chapter I playlist, FINISHED](https://www.youtube.com/playlist?list=PLPn07arnrFILdNx5F10UZRvCBmg127Jua)
[Chapter II playlist, FINISHED](https://www.youtube.com/playlist?list=PLPn07arnrFIItrbcXMJ8ZN6U717_vGCOS)
[Chapter III playlist, FINISHED](https://www.youtube.com/playlist?list=PLPn07arnrFILtK93lmloidbpTjl18qmKy)

</details></details>

**Comment by Tormentor667** [comment]
It will be definitely useful :)

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 Ok, thank you, tomorrow I will start. Here are the in-game settings I will use that differ from the defaults:
<details><summary>Settings</summary>
cl_capfps = 1

BoA menu options:
Weather settings = everything OFF
Effect settings = everything OFF
Rendering settings: distance for 3d objects = 4096
Post Process = everything OFF

Sector Light Mode = Standard
Gamma Correction = 2.0
Texture Filter = OFF
Anisotropic Filter = OFF
</details>

**Comment by Username-N00b-is-not-available** [comment]
Oops, I got a crash after right before going to C1M3:
![temp](https://user-images.githubusercontent.com/59409713/91707361-59a0a000-eb88-11ea-8787-c40743bbd7c9.png)
Commit https://github.com/Realm667/WolfenDoom/commit/faad388c99e08531941c9287e50d42f523afab33, latest gzdoom devbuild.
@AFADoomer, could you please investigate this..?

**Comment by Username-N00b-is-not-available** [comment]
I hoped to marathon through all of BoA today, but now I realize I simply won't manage to do it in time.
I've recorded 9 segments * 15 min = 135 min of footage and uploaded three of them. **I will update the original post and provide the video links there.**

**Comment by AFADoomer** [comment]
Are you on the latest GZDoom?  There were some array pointer serialization changes in recent commits that may help this.  

I may be able to work around this in ZScript later, but that will take some time to chase down.

**Comment by Username-N00b-is-not-available** [comment]
@AFADoomer, as for now no, I had g45-pre-112 x64. Thank you for the information, will start recording C2 with revision 121.

**Comment by Username-N00b-is-not-available** [comment]
Another crash with commit https://github.com/Realm667/WolfenDoom/commit/a0f48a75a6d35178dbbc1d43b9c22a808159582f on revision 121. This time on C2M1 in area just after the purple switch.

**Comment by Username-N00b-is-not-available** [comment]
@AFADoomer, the crashes are very often so I could not even advance in the map, please would you be so kind to find the source of the problem if you have time? Here are some videos of what I've been doing just before the crash:
[crashes.zip](https://github.com/Realm667/WolfenDoom/files/5157840/crashes.zip)

**Comment by Talon1024** [comment]
Is it the same as #278?

**Comment by Username-N00b-is-not-available** [comment]
@Talon1024 Maybe! Though I'm not really sure that they are the same, as mine arose only in very recent revisions (three or four days old), and yours is more than a week.
Next time I'll try to save the crash information.

**Comment by Username-N00b-is-not-available** [comment]
I somehow got through C2M1, M2 and M3, will possibly upload some of them tomorrow. The regular crashes obstruct my walkthrough quite seriously, but I'll try to break through this.

**Comment by AFADoomer** [comment]
The blockingline crash is being addressed in GZDoom...  See thread [here](https://forum.zdoom.org/viewtopic.php?f=2&t=69747).  I'm waiting to see what comes of that before I make any mod-side changes, since I'm pretty sure that none of our code sets or clears the player's blockingline variable anyway.

I've had no luck recreating your other crashes...  I think they may be the same as Talon1024's, but I can't tell without being able to recreate them reliably.

**Comment by Ozymandias81** [comment]
He seems to have deactivate everything, as stated on his Settings spoiler. I still need to find time and patience to play through the whole project, but I tested a week ago C2M1 for 16-20 mins and didn't had any crash. For what I can see N00b also use unlimited sprint too.

**Comment by Username-N00b-is-not-available** [comment]
>unlimited sprint
I really like this feature, but not for this walkthrough. In the Settings spoiler I mentioned **all** the settings I changed after installing a fresh separate copy of gzdoom (except the default control scheme, of course).

**Comment by Username-N00b-is-not-available** [comment]
@AFADoomer, this crash to Windows was on today's devbuild (where the pointer issue was circumvented), it occured when I was crossing the bridge over the first fan area in C2M5_B:
![error](https://user-images.githubusercontent.com/59409713/92141047-95529880-ee1a-11ea-9509-d159406d8067.png)
[CrashReport.zip](https://github.com/Realm667/WolfenDoom/files/5169878/CrashReport.zip)

**Comment by AFADoomer** [comment]
I really can't do anything with engine crashes like this.

What renderer are you using?  OpenGL or Vulkan?

**Comment by Username-N00b-is-not-available** [comment]
Whatever is set by default, in this case OpenGL.
In the meantime I decided to try older revisions, and №80 seems to work just fine and without any crashes. There is a playlist for the second chapter in the first post BTW, and I'm crossing fingers to complete it this weekend.

**Comment by Ozymandias81** [comment]
Can you be so kind and tell us exactly your pc specs @Username-N00b-is-not-available ? Since you are playing with everything off, I guess it is not very performance like in general. Just to be sure from which kind of PC we are seeing those playthroughs.

**Comment by Username-N00b-is-not-available** [comment]
I'm playing with everything off because I don't like most of these effects at all, my opinion is that they only obstruct the view. I consider things like _lens flares_, <strike>white noise</strike> _pixelated film grain_, and especially the goddamn _motion blur_ completely unnecessary to enjoy the levels. Though I do understand the reasons for their presence in the modification, and don't oppose it, since it is so easy for a user to disable them.
As for the performance, my PC is fair enough; I've completed Doom Eternal on it :-) It is a notebook with Geforce MX150 + an integrated card, the CPU is a Core i5-8250U 1.6GHz. It has 8GB RAM. When I'm recording BoA, the battery is being charged, so the energy saving does not affect performance. As far as I know there was no significant lag during these videos?

Current status: slightly more than half of C2 videos uploaded, game completed up to C3M4.

**Comment by Ozymandias81** [comment]
While the CPU should be stronger sadly due to single-threaded code of GZDoom (I suggest something like 3ghz or higher), everything else seems pretty fine. Didn't notice lag on your videos, but indeed playing with everything off something may alter actor behavior, I am not sure though since even I play with many features off. I still need time to check your playthroughs, we must also keep in mind that some things may behave strange because of devbuilds of GZDoom. Will watch them when I can for sure!

**Comment by Username-N00b-is-not-available** [comment]
Looks like the SS Officer uniform seen in C3M5_C recently became bugged: now it does not give the disguise to the player. I found a way to slip through the officer-guarded gates though :-)
Only 4 videos of 40 remain to upload for C2, then it will be finished.

**Comment by Tormentor667** [comment]
What happens when you try that?

**Comment by Ozymandias81** [comment]
Did you load a savegame first? There isn't anything strange within the map, testing it via UDBuilder directly works perfectly, wonder if this script may have some role on the issue you got...
![image](https://user-images.githubusercontent.com/11136108/92294003-5579d700-ef28-11ea-9e0d-0f5b37dde3ef.png)

**Comment by Username-N00b-is-not-available** [comment]
I was playing continuously with saves, and it looks like a DisguiseFailToken somehow was not cleared from my inventory on ending of one of the earlier maps (item 150; looks like I got it in first Mengele battle, I think?) Just removing this item on entering the map seems to be a good enough crutch to work.

<details><summary>Inventory</summary>

]printinv
Inventory for Player 'Player':
    Z_ShadeMe #3484 (1/1)
    BoARedKey #3475 (1/1)
    BoABlueKey #3429 (1/1)
    BoATilt #3423 (1/1)
    Stamina #3410 (100/100)
    UMG43 #3397 (1/1)
    ShakeShaderControl #3377 (1/2147483647)
    Browning5 #3361 (1/1)
    Pyrolight #3352 (1/1)
    NebAmmo #3344 (29/100)
    Nebelwerfer #3343 (1/1)
    Kennkarte #3317 (1/1)
    TrenchShotgun #3274 (1/1)
    Luger9mm #3261 (1/1)
    MineSweeper #3150 (1/1)
    G43 #3062 (1/1)
    Panzerschreck #3060 (1/1)
    KAR98K #2878 (1/1)
    SpyToken #2862 (1/1)
    MP40 #2861 (1/1)
    Firebrand #2859 (1/1)
    Walther9mm #2858 (1/1)
    NaziBerserk #2856 (1/1)
    DeployableMine #2855 (5/5)
    STEN #2854 (1/1)
    LanternPickup #2825 (1/1)
    LanternOil #2824 (2000/12000)
    TeslaCell #2819 (30/200)
    ZyklonMask #2801 (1/1)
    AdrenalineKit #2702 (3/3)
    GrenadePickup #2681 (8/9)
    Cartridge53 #2665 (1/1)
    power #1021 (2343/2400)
    TurretHeatAmmo #569 (0/100)
    TurretBulletAmmo #568 (0/100)
    VitalitySerum200 #556 (20/20)
    Soul #409 (313/9999)
    FlameAmmo #224 (99/175)
    Bandages #173 (3/5)
    12GaugeAmmo #155 (2/32)
    DisguiseFailToken #150 (1/1)
    FieldKit #148 (6/10)
    PanzerAmmo #123 (0/5)
    AktenDreamClue #69 (1/2)
    MauserAmmo #56 (19/40)
    RadioPickup #53 (1/1)
    9MMAMMO #52 (126/256)
    BoACompass #49 (1/1)
    CoinItem #47 (639/9999)
    SandShaderControl #35 (1/2147483647)
    BlurShaderControl #34 (1/2147483647)
    HeatShaderControl #32 (1/2147483647)
    OldVideoShaderControl #29 (1/2147483647)
    BoAUnderwater #21 (1/1)
    BoAVisibility #20 (1/1)
    AstroChaingunLoaded #18 (30/30)
    AstroRocketLauncherLoaded #17 (5/5)
    AstroShotgunLoaded #16 (8/8)
    Browning5Loaded #15 (4/4)
    G43Loaded #14 (10/10)
    Kar98kLoaded #13 (5/5)
    Luger9mmLoaded #12 (8/8)
    MP40Loaded #11 (32/32)
    PanzerschreckLoaded #10 (0/1)
    PyrolightLoaded #9 (50/50)
    StenLoaded #8 (32/32)
    TeslaLoaded #7 (30/30)
    TrenchShotgunLoaded #6 (8/8)
    Walther9mmLoaded #5 (8/8)
    NullWeapon #3 (1/1)
    BasicArmor #1 (3/100)
    HexenArmor #0 (1/1)
  List count: 72

</details>

The officer began shooting me on sight. Though he ceased fire to salute me when I crossed certain linedefs and opened the grate when I showed him the Kennkarte :-) By then I had more than enough ammo to deal with all the troops on the airship, so I did not have to stop recording.

**Comment by Ozymandias81** [comment]
I think I have found the source of the issue on C3M5_C: during the repair sequence, the EngineDestroy script, you'll receive a DisguiseFailToken and that probably gets stored on the inventory and never removed anymore. Problem is how to solve that if we reload the game AFTER the EngineDestroy script while we have removed the DisguiseFailToken with a separate script... Any ideas?
![image](https://user-images.githubusercontent.com/11136108/92307388-59444280-ef96-11ea-9c2f-757262b81181.png)

**Comment by Username-N00b-is-not-available** [comment]
@Ozymandias81 No-no-no, sorry, the problem was in the literal beginning of the map, this script would not have been activated yet. I think the item is most probably just a leftover from previous maps.

Meanwhile I've finished this Blade of Agony walkthrough! There were some 'game-breaking' problems which did not allow intended progression, but thanks to the modification's complexity there were always ways to bypass these (sometimes with the help of other bugs :-) ). This does not mean there is no need to fix these issues anymore :-)
The levels look very impressive, and the combat is nice too. There were frustrating and even pure BS moments from the gameplay perspective (various platforming+instant-death sections, Astrostein yellow key arena), but generally the combat almost always feels good. The enemies have fast reflexes, and they are not that dumb (but there is still some work to accomplish on this front...); weapons are satisfying (I couldn't find a use for the Tesla Cannon. maybe should have confronted the flying piggy with it), my favourite one is Kar98k in unscoped mode.
Boss battles are also good, but what should the player do if they run out of souls in the last part of C3M6_B?
Also, Hitler feels pretty helpless if you have enough souls on you, as I did in my playthrough. He went from 'full health' to 'corpse' in 22 seconds, most of which he was invulnerable due to 'changing his phase'. Maybe giving him 100K health (and possibly adding 4 phases instead of 2, with lower transition times) could improve the situation? :-) For example, we can make him also throw packets of Zyklon grenades (which I recently drew) in clusters, and also some swastika/Thule symbol whirlwinds and whatnot. This way we will need some health supply for the player.
For me C2 was the most difficult and lengthy episode, but I might be biased since I basically had to pistol-start once a couple of missions and could not use saves during some.
All the levels are very aesthetically pleasing.

There are 19 videos in the C3 playlist for now, I'll upload the remaining ones ASAP (not now because I hit Youtube's daily video limit) and send a message when I'll have it done.

**Comment by Ozymandias81** [comment]
Well [uploaded](https://github.com/Realm667/WolfenDoom/commit/3a2eb728c62ad1e8641d23d99cc4784a1e5150dc) another pack of fixes for C3m5_C, maybe you can retest C3 whenever you can and see if that happens again.

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 @Ozymandias81 @AFADoomer @Talon1024
The full game playthrough has been uploaded to Youtube, see links in the first post.
I've uploaded the videos there so as for you to watch them in 2x or at least 1.5x speed for the reasons of productivity (there are 14+40+28=82 parts, each 10~15 minutes long). I did not aim for 100%, but tried to showcase what a generic BoA playthrough would consist of, and to draw attention to moments I found important. Again, I hope that this effort was not wasted, so please watch the videos when you have time and patience to do it.

**Comment by Tormentor667** [comment]
Thanks kindly for that effort. I will watch these in the next days in my midday breaks and evenings

**Comment by Username-N00b-is-not-available** [comment]
Gee, I wonder how many comments the developers have left under my playthrough during these three weeks, must be a lot...
*checks youtube video page*
![views](https://user-images.githubusercontent.com/59409713/94270240-6f677200-ff48-11ea-951b-f8c968fdcafa.png)
Zero comments.
I know that you all are busy, I have very little time myself now, but I would like to remind you of this. Please do not forget to actually skim through these videos BEFORE the release; as far as I can remember, I've caught a few dozens of bugs during the recording, and it looks like most of them are present now.

**Comment by Tormentor667** [comment]
Don’t worry, I didn’t miss that, I just didn’t have time yet to watch them. Office is currently a pain and I simply didn’t have time yet to watch these. But I WILL! 👍

**Comment by DoomJedi** [comment]
Yes, I am sure guys will watch and fix all before any release candidate

**Comment by Tormentor667** [comment]
Making good progress with watching the vids - currently at C1M5

**Comment by Tormentor667** [comment]
**Done with Chapter 1**

**Comment by Ozymandias81** [comment]
Be always careful making changes to maps since we are working over devbuilds of GZDoom and not stable builds. Also many effects and stuff are deactivated so this could also have a role on some issues, yet still didn't had time to check videos (and next 2 months will be probably like that unfortunately, Torm knows why on my end). But I'll do my best to watch them when I can

**Comment by Tormentor667** [comment]
Continuing with C2 now :)

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 Beware, this will be a lot more tough to watch than C1...

**Comment by Tormentor667** [comment]
...because you die more often? ;)

**Comment by Username-N00b-is-not-available** [comment]
WAY more often, especially in later parts of C2M1.

**Comment by Tormentor667** [comment]
I already notice that :) Why didn't you use "resurrect" or save more often?

**Comment by Username-N00b-is-not-available** [comment]
- resurrect: I played on Mein Leben which has sv_cheats 0.
- save more often: I tried my best to do this, but there were consistent crashes right upon saving in some areas.

So I was out of luck this time. And no, lowering the skill level was not an option :-)

**Comment by Tormentor667** [comment]
Anyway, I will watch it till the end, no worries :)

**Comment by Tormentor667** [comment]
**Done with Chapter 2**, will continue with Chapter 3 now.

**Comment by Tormentor667** [comment]
Watched all of them and posted everything that's needed (or fixed it), so thanks for the playthrough man :)

**Comment by Username-N00b-is-not-available** [comment]
↑↑↑ Resurrecting the issue ![resurrect](https://user-images.githubusercontent.com/59409713/102812424-09b84400-43d8-11eb-89c3-1687e7ead574.png) ↑↑↑
---
@Tormentor667 @Ozymandias81 @Talon1024

I have recorded another full playthrough. This time I played on the lowest skill and saved extensively (since the crashes have been apparently dealt with), so it is shorter than the previous edition (11 parts vs 14 on C1, 26 vs 40 on C2, and have not yet counted C3). I used GZDoom 4.6pre20 and this BoA commit: https://github.com/Realm667/WolfenDoom/commit/518865ee16c6d952edc283cc9723088c018c635e

Hopefully I will have it fully uploaded in a few days. For now, there is only the first episode. If you do decide to watch some of these, please use the speed multipliers, because I'm not a speedrunner, and the footage could become boring to watch.

Overall I think the game is also fine on this skill, there are still many nitpicky problems on maps here and there (including some very old ones), but everything is playable, finishable and fun to blast through as a power fantasy. What stood out from this list of problems in my mind:

- [ ] The screen does not fade from black to normal upon finishing some of the cutscenes (e. g. before C2M0_A).

- [ ] Multiple problems on C3M0_B, including those which were probably fixed in the incoming commits, but also recurring 'once-fixed' ones, like the inventory carryover (from INTERMAP to this map), or an escape route through the main door (as the scripts do not manage to be activated in time), or the inability to exit the level.

- [ ] The unfinished zombie chamber is useless in its current state, being cheesable very easily. I have indicated some thoughts on balancing it in the video.

- [ ] The keys' carryover during transit from C3M6_A to _B.

- [ ] The green key secret area again blocks the player from entering it. Is that area meant as the super secret BTW, or some other one? Also, to @Ozymandias81 who is about to start fixing the objectives on the map: they do not work here at all.

- [ ] In the later parts of the game my inventory essentially became "a women's purse" with a lot of useful items, but no way to quickly find the exact one that you need at the current moment. For example, after killing Hitler I had 17 types of usable items in the inventory, and there is no way to order them properly in a way that would be convenient for the specific player, you have to scroll through the entire list to find one. I have two ideas, any of them would involve a lot of work, but here they are: we either can make a inventory screen which could be controlled by mouse, or just arrange the items which are currently present in a 2D grid and add two buttons whichcontrol the vertical movement through the list. I do not consider adding separate keys for each inventory type, since there is simply not enough of them. Maybe something more simple could work here, I would like to hear your suggestions, if you have any, @AFADoomer, @Talon1024 and @Ozymandias81.

- [ ] To be frank, the Commander Keen levels were really boring and tiresome for me. The complicated portal structures and lack of proper automap indication of doors for the keys of each colour made me use the editor multiple times to make some progress there, except on the first level which I have already played through multiple times to this time (it also suffers from this switchhunting). It is not always clear what some of the switches do. Also, the 12 inchworms in the third level do not activate the exit sequence, and there are no Cyrillic-specific letters in the floating princess' font.

- [ ] In his final battle Hitler disappeared again. I seriously do not know what could have been the cause of this, played a bit with the settings after encountering this, but could not make Hitler show himself. He should also be instantly awakened after spawning and pointed at the player (as for now, he is not).

Video links
---

<details><summary>Chapter I</summary>

https://www.youtube.com/playlist?list=PLPn07arnrFIKme92teVJRsJRwO48-oruc

</details>

**Comment by Tormentor667** [comment]
Thanks kindly for your effort, I will watch the videos during holidays :)

**Comment by Username-N00b-is-not-available** [comment]
I hope the speed of my Internet connection will allow it :-)

**Comment by DoomJedi** [comment]
About Keen levels. 
Well, Keen gameplay is Keen gameplay, for better and for worse.
The only thing that we can improve - is indeed to simplify portal structure and make more obvious key to door, switch to switching target routes.

Maybe to use Hexen type text hints for switches? "You hear elevator starting moving in a distance", "You hear a door opening" etc? Maybe with actual "sound" plays related to elevator, platform or door opening?

**Comment by Tormentor667** [comment]
> Maybe to use Hexen type text hints for switches? "You hear elevator starting moving in a distance", "You hear a door opening" etc? Maybe with actual "sound" plays related to elevator, platform or door opening?
Please not. This is not how Keen works. It's "search and discover", and it's a perfect replica of how Keen worked. We won't change this.

**Comment by DoomJedi** [comment]
This is indeed how Keen works, alot of exploration. A bit more confusing now in 3D, but still.

**Comment by Username-N00b-is-not-available** [comment]
@DoomJedi I agree on the gameplay progression, maybe it's just not my type. I enjoyed the artwork, the level design and the platforming very much, they are excellent. It's only the presentation which made it boring for me.
I doubt that it'd be possible to significantly simplify the layout of portals without scrapping some areas. Instead, I would like to suggest doing something with the automap, it is utterly useless in this levels now, since the maps are mostly open areas with the platforms or paths having similar-styled shapes.
As an experiment, could anyone who has played SM03 (including @Tormentor667 who had built it) please tell me what area this portal leads to without opening either the game or editor? Bonus points is what key you have to use just before accessing that area.
![Screenshot_Doom_20201222_153629](https://user-images.githubusercontent.com/59409713/102889461-c0193900-446b-11eb-97de-8a4dcfbef889.png)
@Tormentor667 "Search and discover" is fine, and such messages would be out of place, but the three levels are not only three-dimensional,  they are also extensively non-Euclidean.

**Comment by Username-N00b-is-not-available** [comment]
All videos uploaded, posting the links. There are 63 video parts, total length is around 160 min for C1, 340 min for C2 and 360 for C3. AFAIK I haven't missed any maps, but that could be possible, please tell me if it turns out to be so.

<details><summary>Links</summary>

Chapter I: https://www.youtube.com/playlist?list=PLPn07arnrFIKme92teVJRsJRwO48-oruc
Chapter II: https://www.youtube.com/playlist?list=PLPn07arnrFIIFu_YFG4CWvJ01MnNNTvIU
Chapter III: https://www.youtube.com/playlist?list=PLPn07arnrFIIsPcWrdTmb1fwbK5Z9B-ny
</details>

Happy holidays to everyone ([including](https://github.com/Realm667/WolfenDoom/issues/281#issuecomment-749197829) @Tormentor667 :-D)!

**Comment by Tormentor667** [comment]
> The green key secret area in C3M6_B again blocks the player from entering it. Is that area meant as the super secret BTW, or some other one? Also, to @Ozymandias81 who is about to start fixing the objectives on the map: they do not work here at all.

@Username-N00b-is-not-available can you elaborate this a bit more in detail?

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 I can't reproduce the secret area bug now for some reason, works fine... Though the structure of the map near (-1600, 5472) suggests that there could be some problem with the geometry (multiple nonconnected sectors are joined into one) because of previous level editing sessions.
BTW you have not answered, is the Realms of Arkania area the super secret one?

**Comment by Tormentor667** [comment]
@Username-N00b-is-not-available - I fixed the geometry just not, uploading a new file. Regarding the "super secret one", I never declared it to be or either used the term but yes, it's a cool secret :)

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 I thought you wrote that line in the objectives for C3M6_B (which have not been tied to level triggers yet), but it turns out it was Ozy's work. Sorry :-)

**Comment by Username-N00b-is-not-available** [comment]
Next portion of issues which need help (resulting from @Tormentor667 watching my videos):

- [ ] @Talon1024 There are some missing characters (Cyrillic ones and probably dashes) to the font used by the Commander Keen dialogues, could you add them please?
- [ ] @AFADoomer Could you please make the treasure counter on the automap on by default?
- [ ] @AFADoomer The smoke monster boss has its healthbar shown incorrectly on skill 0: https://youtu.be/UXj-FksElss?t=550
- [ ] @AFADoomer Enemy soldiers try to shout an active sound every tic (old bug which never got fixed): https://youtu.be/zMulvlANoQ0?t=430

**Comment by Tormentor667** [comment]
>     * @AFADoomer The smoke monster boss has its healthbar shown incorrectly on skill 0: https://youtu.be/UXj-FksElss?t=550

I can't reproduce this, it works for me. Can you try that again?

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 Here you are. It is important to test on skill 0:

https://user-images.githubusercontent.com/59409713/103156649-d270db00-47bb-11eb-8a2c-a369d0288ba1.mp4

**Comment by Tormentor667** [comment]
Okay this needs to be investigated by @AFADoomer I guess.

**Comment by Tormentor667** [comment]
Done with C2, will continue with C3 soon!

**Comment by Tormentor667** [comment]
I am done with C3, watched it all, fixed what I could, commented what I could, assigned what I could. The remaining stuff is mostly programming, so some of the coders need to look at them. I also moved the remaining issues to the first post.

Thanks a lot @Username-N00b-is-not-available for this brilliant work, it was really helpful!

**Comment by Tormentor667** [comment]
Marked a few things with fixes, assigned some issues and also commented some others.

**Comment by DoomJedi** [comment]
"There is a SKLKC0.png placeholder key sprite which I have made, could you please improve or just redo it completely?"
What is the key visual concept? Is it still skull-related, or should have a different look?

**Comment by Username-N00b-is-not-available** [comment]
@DoomJedi It should just be a large key for a padlock, it is used for a storage door secret in C3M5_A. It is not related to skulls or anything else occult. I wanted to base it on the skull key graphic to make it stand out from the usual keys.

**Comment by DoomJedi** [comment]
Okay, will be on it later.

    On Friday, January 1, 2021, 4:15:36 PM GMT+2, Username-N00b-is-not-available <notifications@github.com> wrote:  
 
 


@DoomJedi It should just be a large key for a padlock, it is used for a storage door secret in C3M5_A. It is not related to skulls or anything else occult. I wanted to base it on the skull key graphic to make it stand out from the usual keys.

**Comment by Tormentor667** [comment]
@Username-N00b-is-not-available should I do another take on the zombie chamber? I think I have some ideas how this could work.

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 Of course, please make the changes!
BTW I've read your message about actors for CK map portals, and I'm wondering about the possibility to implement this, since disjoint portal areas are drawn by default on different 'canvases'. We can change ```am_portaloverlay``` to 0, but this will also affect other maps.
Also the colours on some of the doors which you have added only seem to appear on the automap after an 'iddt', and some ones like the blue door on (11900, 440, -400) of SM01 are marked with another (misleading) colour.

**Comment by Tormentor667** [comment]
Fixed the door colors and you are right when it comes to the canvas’ - I’d say we keep it as is then

**Comment by Tormentor667** [comment]
> The screen does not fade from black to normal upon finishing some of the cutscenes (e. g. before C2M0_A). [assigned to @Ozymandias81 @Talon1024]

I tried introduction movies to C2M0_A and C2M6_B as well as CUTSCEN1-3 maps, I wasn't able to reproduce this so I guess it's fixed.

**Comment by Username-N00b-is-not-available** [comment]
I decided to do some thorough testing again (but without videos). For now I have finished the first two chapters; I hope this will be useful (and I can only wish you best of luck in fixing them). Third chapter is probably coming in a few days, I'd like to take a break...
![GODDAMN_WHEN_WILL_HE_STOP_IT](https://user-images.githubusercontent.com/59409713/105547297-6ed0c380-5d0f-11eb-8a2e-3a4ad3a0f224.png)

**Comment by Tormentor667** [comment]
I will do a full run in German as soon as I am done with my translations :) thanks again for your effort!

**Comment by Username-N00b-is-not-available** [comment]
Now that I have again finished playing the third episode, the original post now contains the full supply of bugs I've caught with my net up until today. Time for me to refrain from testing for some time...

**Comment by Tormentor667** [comment]
Regarding some issues:

- [ ] Skipping introduction letter: Works for me, tried it standalone and as mod
- [ ] Mugshots talking being too fast: Also tried it as mod and standlone, works fine for me
- [ ] Nemesis timing at the end of C3M0_A: Has been adjusted by @Talon1024 and @AFADoomer 
- [ ] C2M3 Cyan door: Can't be fixed due to the nature of polyobjects (though tried something)
- [ ] CUTSCEN2: I can't reproduce this (wether on warping to the map or through a playthrough) so I marked that as fixed
- [ ] NoAutosveHint: Has been added by me
- [ ] ArmorShards: Don't give item bonus scores anymore
- [ ] C3M0_A House windows (-4753, 5531, -184): Breakable windows can't be fixed because of their midtexture flagging, we simply need to live with that

Will continue with fixes soon.

**Comment by Tormentor667** [comment]
Fixed a few more bugs, I hope, @Ozymandias81 @AFADoomer @Talon1024 and @Username-N00b-is-not-available are joining me here to fix up the rest. Some of them are out of my scope.

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 I'd be happy to keep helping you, but the spring semester in Russia starts exactly tomorrow, so my activity here will sadly become quite limited... I hope to show up anyway and throw in my two cents every now and then.

**Comment by Tormentor667** [comment]
I fully understand that. But if you have some free time to help us out, we'd be all very happy :)

**Comment by Username-N00b-is-not-available** [comment]
Duplicating the unresolved questions & suggestions which need **discussion** (not plain old bugs) in a separate post, so it would be possible to read them through. Please share your thoughts on them, some are quite important:

- C1M5: there is absolutely no punishment for the Nebelwerfer being destroyed, I can Panzerschreck it during wave 8 and still get all credit for the mission. Maybe a $250 fine would do?
- Commander Keen messages can be skipped by any key including the movement ones, possibly should only leave 'use' for this.
> This matches every other fullscreen message/menu interface in the game right now.  Is the real problem that message pages are accidentally being skipped? - AFA (23 Jan)

> Yes, you are right, that's what I meant. --N00b, 27 Jan

- On normal skill, please remove target icons which are shown on snipers. They are not hitscan, and the player can already see the bullet trails.
- Suggestion: show the mission summary screen (the one which is in the briefing room in the HQ) after a chapter playthrough?
- Lt. Heit shows a message about the player completing Schwarzwald before C2M3.

> From what I can see, Randi is supposed to use lines BARRANDICONV01 through 06 before C2M2, then nothing new before C2M3, then 07 before C2M4, 09 before M5, 10 before M0, and finally 11 before M6 if M0 was completed. BARRANDICONV08 about "the Thor's Hammer and the long d*ng" does not have a place to fit in. --N00b
- BARRANDICONV11 is never spoken by him (after completing Astrostein he still has his 10th message).
- C2M4 suggestion: add a reward for destroying all planes?
- One of the textures used in C2M6_C has 'Projekt Seeteufel' on it, maybe it is better to either rename the enemy or change the project name to https://de.wikipedia.org/wiki/Schwertwal_(U-Boot) so as not to confuse the players?
- C3M0_A: killcount is totally off, it may be better to disable showing it at all. Pre-placed zombies count towards kills, spawned ones not, and it is impossible for a player to distinguish them.
- C3M0_A: we probably should use SpyGive-like behaviour for bonus items (drop the surplus items near the player). That will likely mean migration from ACS.
- C3M0_A: absolutely nothing indicates that to start assaulting the SS HQ you have to gather all three weapons (Browning, G43, Nebelwerfer) from their specific spots. Else the player can only wonder what to do after grabbing two keys and, for example, missing the butcher's with the rifle. I think the corresponding primary objective should be re-added.
- C3M0_A: Is it really necessary for the player to wait several minutes until the Nemesis wakes up again after destroying the SS HQ? Could you please force Nemesis to revive himself when the player leaves the area, or at least give another message the player to wait a bit 'until the reinforcements arrive'?
- C3M0_A: I have severe lag in the city part of the level, I have to pause the game every few seconds to avoid stuttering. Could anyone please think of some measures to reduce the workload on the CPU, maybe adding more one-sided linedefs which block the view, or replacing lots of small decorations with bigger ones?
- C3M2: the map is lagging as hell, including during the machinegun hill defending scene and the tank ambush (where it is especially frustrating), both on default and low settings. On default my FPS does not raise over 15 and frequently drops to 5. Any measures here as well?
- Medics are useless, they get activated only when the player comes close (and shoots them, of course), they need to roam more freely to be a threat (this is related both to map placement and actor behavior). This is not only true of C3M2, but all maps which use medics.
- C3M2: On default settings, because of the post-processing effects the tank ambush plays awful. You can't see the Pz. IV's properly because of that shaders and BS fog, not to mention that they cause additional workload on the GPU when it's already lagging excessively. Also the Pz. IV's have more health than Tigers (but still go down in two main cannon shots).
- Would it be appropriate to remove the 'Operation' word from C1M1 through C3M0_B entries of the CSV (they hold the map names)? Some of them are too long and overlap with the playing time in the INTERMAP stats screen.
- Is it really necessary to have a damage penalty for unscoped Kar98K fire?
- C3M4: please remove countkill on the pre-placed zombies. There are no problems and you can get 100%, but to a first-time player it would be very non-obvious that the zombies on the ramp across the river count towards kills.
- C3M6_B: Hitler does not count towards kills. Or should he?
- If the player has no souls, he is able to power up the Firebrand with his own HP. That's fine, but this way they are still able to use the altfire attack, and it would cost them nothing until the effect runs off and the player has to use up their HP again. Is this tolerable or not?

---
```summon``` @Tormentor667, @Ozymandias81, @AFADoomer, @Talon1024

**Comment by Tormentor667** [comment]
Closing this, as I have opened this: https://github.com/Realm667/WolfenDoom/issues/383
